ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tx-queues.j2

```diff
@@ -0,0 +1,43 @@
+{# eos - tx-queue profiles #}
+   !
+   tx-queue {{ tx_queue.id }}
+{% if tx_queue.bandwidth_percent is arista.avd.defined %}
+      bandwidth percent {{ tx_queue.bandwidth_percent }}
+{% elif tx_queue.bandwidth_guaranteed_percent is arista.avd.defined %}
+      bandwidth guaranteed percent {{ tx_queue.bandwidth_guaranteed_percent }}
+{% endif %}
+{% if tx_queue.priority is arista.avd.defined %}
+      {{ tx_queue.priority }}
+{% endif %}
+{% if tx_queue.shape.rate is arista.avd.defined %}
+      shape rate {{ tx_queue.shape.rate }}
+{% endif %}
+{% if tx_queue.random_detect is arista.avd.defined %}
+{%     if tx_queue.random_detect.ecn.count is arista.avd.defined(True) %}
+      random-detect ecn count
+{%     endif %}
+{%     if tx_queue.random_detect.ecn.threshold is arista.avd.defined %}
+{%         set units = tx_queue.random_detect.ecn.threshold.units %}
+{%         if tx_queue.random_detect.ecn.threshold.min is arista.avd.defined %}
+{%             set min_threshold = " minimum-threshold " ~ tx_queue.random_detect.ecn.threshold.min ~ " " ~ units  %}
+{%         else %}
+{%             set min_threshold = "" %}
+{%         endif %}
+{%         if tx_queue.random_detect.ecn.threshold.max is arista.avd.defined %}
+{%             set max_threshold = " maximum-threshold " ~ tx_queue.random_detect.ecn.threshold.max ~ " " ~ units %}
+{%         else %}
+{%             set max_threshold = "" %}
+{%         endif %}
+{%         if tx_queue.random_detect.ecn.threshold.max_probability is arista.avd.defined %}
+{%             set max_probability = " max-mark-probability " ~ tx_queue.random_detect.ecn.threshold.max_probability %}
+{%         else %}
+{%             set max_probability = "" %}
+{%         endif %}
+{%         if tx_queue.random_detect.ecn.threshold.weight is arista.avd.defined %}
+{%             set ecn_weight = " weight " ~ tx_queue.random_detect.ecn.threshold.weight %}
+{%         else %}
+{%             set ecn_weight = "" %}
+{%         endif %}
+      random-detect ecn{{ min_threshold }}{{ max_threshold }}{{ max_probability }}{{ ecn_weight }}
+{%     endif %}
```

Comment:
First see my connect on using the presence of the dict alone under the doc template.
Next, please look at other templates, on how we add CLI tokens bit by bit to _one_ variable instead of creating multiple like here.
Ex. (please adjust of course)
```j2
{% if feature.enabled is arista.avd.defined(true) %}
{%     set feature_cli = "basic command" %} <-- notice not space after the command
{%     if feature.knob1 is arista.avd.defined %}
{%         set feature_cli = feature_cli ~ " knob1 " ~ feature.knob1 %} <-- notice the leading space before the keyword, and no trailing space.
{%     endif %}
... more knobs ...
      {{ feature_cli }} <--- Just printing the CLI var which may or may not contain extra tokens.
{% endif %}
```

Reply:
Thank you for pointing this out! I agree it is more elegant. Refactored both `tx-queues.j2` and `uc-tx-queues.j2`